meshers/blocky/voxel.cpp

```diff
@@ -230,12 +230,20 @@ static void bake_cube_geometry(Voxel &config, Voxel::BakedData &baked_data, int
 		for (unsigned int i = 0; i < 4; ++i) {
 			uvs[i] = (config.get_cube_tile(side) + uv[i]) * s;
 		}
+
+		if(bake_tangents){
+			std::vector<float> &tangents = baked_data.model.side_tangents[side];
+			for (unsigned int i = 0; i < 12; ++i) {
```

Comment:
Where does the 12 come from?
This is going to add 12 sets of tangents per side, is that wanted?

Reply:
It's been awhile since I looked at this but I'm pretty sure this was intentional. The visual server(visual_server.cpp ~line 500) expects 4x the number of vertices for tangents. The vertex array is received as an array of uint8_t instead of Vector3 so we have to consider the flattened size. 

For each side made up of 4 vertices, there will be 12 floats. So I added one tangent per each float(since the tangent is 4 floats). 

I don't remember if there was any deeper reason based on research. It might honestly be wrong.

Reply:
I thought there should be 4 floats per vertex (3 for tangent, 1 for bitangent?). Which means it would add up to 16 floats per side.
However this `for` is nested. 12 * 4 means 48 floats, and this is already within a third `for` iterating over each side, si it doesn't sound right. I wonder what `i` or `12` even is, maybe it should just not be there?

`VisualServer` deals with `uint8_t` at some point but that's only for compression or data conversion before it sends it to OpenGL, I don't think it's relevant here.

Reply:
Sorry for resurrecting this. Been busy.

I did some testing and I think you're right. When I change it to 4 it works fine. I'll patch it and commit in a bit.